RELEASING.md

```diff
@@ -0,0 +1,57 @@
+# How to release a new git.gem
+
+Releasing a new version of the `git` gem requires these steps:
+  * [Prepare the release](#prepare-the-release)
+  * [Create a GitHub release](#create-a-github-release)
+  * [Build and release the gem](#build-and-release-the-gem)
+
+These instructions use an example where the current release version is `1.5.0`
+and the new release version to be created is `1.6.0.pre1`.
+
+## Prepare the release
+
+From a fork of ruby-git, create a PR containing changes to (1) bump the
+version number, (2) update the CHANGELOG.md, and (3) tag the release.
+
+  * Bump the version number in lib/git/version.rb following [Semantic Versioning](https://semver.org)
+    guidelines
+  * Add a link in CHANGELOG.md to the release tag which will be created later
+    in this guide
+  * Create a new tag using [git-extras](https://github.com/tj/git-extras/blob/master/Commands.md#git-release)
+    `git release` command
+    * For example: `git release v1.6.0.pre1`
```

Comment:
Should the tag really be made in the actual PR? This will work but only as long as we are strict with using `--ff-only`-style merging (fast forward merges without merge commits).

The other option would be to tag it once the PR is merged instead.